products/Product.js

```diff
@@ -900,6 +907,43 @@ export default class Product {
     return this.getAvailabilityDateRanges().getDateList(type)
   }
 
+  /**
+   * Checks, if a product definition of this product has the full time span within validity dates and availability ranges
+   * @param productDefinitionInstance
+   * @param bookingStartInstance
+   * @return {boolean}
+   */
+  checkAvailableDates(productDefinitionInstance, bookingStartInstance) {
+    // get an list of available dates of the product
+    const availableDateList = this.getAvailableDates('date')
+    // get a list of dates of the product definition (booking date + duration days)
+    const durationDays = productDefinitionInstance.getDurationDays()
+    const endDate = new Date(
+      bookingStartInstance.getFullYear(),
+      bookingStartInstance.getMonth(),
+      bookingStartInstance.getDate() + durationDays - 1,
+      0,
+      0,
+      0,
+      0
+    )
```

Comment:
why not use moment to parse the date?

Reply:
Because built-in JS-Date is faster. And because I need endDate to be a JS-Date-Object